modules/social_features/social_embed/src/Plugin/Filter/SocialEmbedConvertUrlToEmbedFilter.php

```diff
@@ -93,154 +83,239 @@ public function process($text, $langcode) {
+      }
+    }
+
+    // If there is no text at this point revert to the previous text.
+    return strlen((string) $text) > 0 ? $text : $saved_text;
+  }
+
+  /**
+   * Makes links out of absolute URLs.
+   *
+   * Callback for preg_replace_callback() within convertUrls().
+   */
+  public static function socialFullUrlToEmbed(array $match): string {
+    return self::socialUrlToEmbed($match);
+  }
+
+  /**
+   * Makes links out of domain names starting with "www.".
+   *
+   * Callback for preg_replace_callback() within convertUrls().
+   */
+  public static function socialPartialUrlToEmbed(array $match): string {
+    return self::socialUrlToEmbed($match, 'https://');
+  }
+
+  /**
+   * Makes links out of domain names starting with domain name.
+   *
+   * Callback for preg_replace_callback() within convertUrls().
+   */
+  public static function socialShortUrlToEmbed(array $match): string {
+    return self::socialPartialUrlToEmbed($match);
+  }
+
+  /**
+   * Process URL for embedding.
+   *
+   * @param array $match
+   *   The array with matched strings.
+   * @param string $full_url_prefix
+   *   The internet protocol
+   *   Send your value (https:// or https://www.) for partial and short links.
+   *
+   * @return string
+   *   Processed link.
+   */
+  public static function socialUrlToEmbed(array $match, string $full_url_prefix = ''): string {
+    try {
+      $url_for_processing = $match[1];
+      $social_embed_helper = \Drupal::service('social_embed.helper_service');
+      // Decode URL.
+      $url_for_processing = Html::decodeEntities($url_for_processing);
+      // Default URL for return.
+      $result_link = $url_for_processing;
+      // Full URL with protocol (http, https etc.).
+      $full_url = $full_url_prefix . $url_for_processing;
+      $info = \Drupal::service('url_embed')->getUrlInfo($full_url);
+
+      if ($info) {
+        /** @var \Drupal\user\Entity\User $user */
+        $user = \Drupal::currentUser()->isAnonymous() ? NULL : User::load(\Drupal::currentUser()->id());
+        $embed_settings = \Drupal::configFactory()->get('social_embed.settings');
+
+        if (!empty($info['code'])
+          && (($user instanceof User
+              && $user->hasField('field_user_embed_content_consent')
+              && !empty($user->get('field_user_embed_content_consent')->getValue()[0]['value'])
+              && $embed_settings->get('embed_consent_settings_lu'))
+            || ($user == NULL && !empty($embed_settings->get('embed_consent_settings_an')))
+          )
+        ) {
+          // Replace URL with consent button.
+          return $social_embed_helper->getPlaceholderMarkupForProvider($info['providerName'], $full_url);
+        }
+        else {
+          $white_list = $social_embed_helper->getPatterns();
+
+          // Check if the URL provided is from a whitelisted site.
+          foreach ($white_list as $item) {
+            // Testing pattern.
+            $testing_pattern = '/' . $item . '/';
+
```

Comment:
```suggestion
  // For "facebook" and "instagram" links embedding require to set up an application.
  // Sometimes it doesn't have a sake to do it. Let's return a link without embedding
  // if the application isn't connected.
  if (
    preg_match("/facebook.com\/|instagram.com\//i", $result_link) &&
    !\Drupal::config('url_embed.settings')->get('facebook_app_id')
  ) {
    return $result_link;
  }
```

Reply:
To avoid instagram and facebook links to be just a text:
<img width="698" alt="Screenshot 2024-02-27 at 14 05 31" src="https://github.com/goalgorilla/open_social/assets/19921926/a1a8f9c3-2583-4998-b25d-b9b5553f7237">
I would add this snippet to have those as a links if application isn't setup:
<img width="698" alt="Screenshot 2024-02-27 at 14 03 36" src="https://github.com/goalgorilla/open_social/assets/19921926/c700e831-dc3e-470b-a3a7-57719155ec08">